crates/net/common/src/bandwidth_meter.rs

```diff
@@ -24,65 +24,16 @@
 
 use crate::stream::HasRemoteAddr;
 use std::{
-    convert::TryFrom as _,
     io,
     net::SocketAddr,
     pin::Pin,
-    sync::{
-        atomic::{AtomicU64, Ordering},
-        Arc,
-    },
     task::{ready, Context, Poll},
 };
 use tokio::{
     io::{AsyncRead, AsyncWrite, ReadBuf},
     net::TcpStream,
 };
 
-/// Meters bandwidth usage of streams
-#[derive(Debug)]
-struct BandwidthMeterInner {
-    /// Measures the number of inbound packets
-    inbound: AtomicU64,
-    /// Measures the number of outbound packets
-    outbound: AtomicU64,
-}
-
-/// Public shareable struct used for getting bandwidth metering info
-#[derive(Clone, Debug)]
-pub struct BandwidthMeter {
-    inner: Arc<BandwidthMeterInner>,
-}
-
-impl BandwidthMeter {
-    /// Returns the total number of bytes that have been downloaded on all the streams.
-    ///
-    /// > **Note**: This method is by design subject to race conditions. The returned value should
-    /// > only ever be used for statistics purposes.
-    pub fn total_inbound(&self) -> u64 {
-        self.inner.inbound.load(Ordering::Relaxed)
-    }
-
-    /// Returns the total number of bytes that have been uploaded on all the streams.
-    ///
-    /// > **Note**: This method is by design subject to race conditions. The returned value should
-    /// > only ever be used for statistics purposes.
-    pub fn total_outbound(&self) -> u64 {
-        self.inner.outbound.load(Ordering::Relaxed)
-    }
-}
-
-impl Default for BandwidthMeter {
-    fn default() -> Self {
-        Self {
-            inner: Arc::new(BandwidthMeterInner {
-                inbound: AtomicU64::new(0),
-                outbound: AtomicU64::new(0),
-            }),
-        }
-    }
-}
-
 /// Wraps around a single stream that implements [`AsyncRead`] + [`AsyncWrite`] and meters the
 /// bandwidth through it
 #[derive(Debug)]
```

Comment:
this type is now just a wrapper+delegate and now effectively useless.

let's remove this file entirely

Reply:
> this type is now just a wrapper+delegate and now effectively useless.
> 
> let's remove this file entirely

not to misunderstand, I should completely remove `MeteredStream`?

Reply:
yep, this type no longer does anything useful

Reply:
@mattsse 
```rust
/// The type of the underlying peer network connection.
pub type EthPeerConnection = EthStream<P2PStream<ECIESStream<MeteredStream<TcpStream>>>>;

/// Various connection types that at least support the ETH protocol.
pub type EthSatelliteConnection =
    RlpxSatelliteStream<ECIESStream<MeteredStream<TcpStream>>, EthStream<ProtocolProxy>>;
```
what to do with these types which utilizes `MeteredStream`?

Reply:
I think @mattsse, this is done.